voluncheer/profiles/tests/test_views.py

```diff
@@ -263,3 +264,23 @@ def test_volunteer_can_see_past_opportunities(self):
             self.opp.attended_volunteers.remove(self.vol)
             response = self.client.get(reverse("saved_events"))
             self.assertNotIn(self.opp, response.context["opportunity_attended"])
+
+    def test_badge_progression(self):
+        with self.subTest("hours_until_first_badge"):
+            self.vol.hours_volunteered = datetime.timedelta(hours=10)
+            progress, hours_remaining, _ = profile.badge_progression(self.vol)
+            self.assertEqual(progress, 20)
```

Comment:
It was unclear that progress was percentage until reading the rest of the code, especially back to back with hours_remaining. Perhaps a docstring could elaborate that the test is for returning correct hours remaining until next badge and progress percentage?

Reply:
I've added a bit more detail to the implementation. WDYT?